### PR TITLE
backend: update coin control outputs sorting

### PR DESCRIPTION
**Pull request by strmci**
Currently, the list of all UTXOs in the coin control selection is ordered by amount. Users sometimes re-use addresses and are not aware that there might be other UTXOs they didn't want or want to use on the same address as their selected output. Grouping the UTXOs by address helps the user to detect if there are other UTXOs.

This commit updates the sorting of outputs in the coin control dialog and is now grouped by address from the biggest to the lowest amount.

### CODE REVIEW

**Comment by strmci** [comment]
@thisconnect thanks, I updated the commit message, PTAL

**Comment by thisconnect** [comment]
Before this change the sorting was outputs by amounts. Now with this change it is outputs grouped by address then sorted by amounts.

For users that didn't re-use addresses nothing should change.

For users that did reuse address the list should change so that all UTXO's of the same address are shown together.

I think it would be great if there is a banner on top that shows "Warning: some addresses were re-used" or similar (banner of shame).

File: backend/coins/btc/account.go [hunk 1]
```diff
@@ -773,19 +772,55 @@ func (account *Account) CanVerifyAddresses() (bool, bool, error) {
...
+		sum     int64
+	}, 0, len(grouped))
+
+	// Calculate sums of values for each group and store in the sums slice
+	for address, outputs := range grouped {
+		var sum int64
+		for _, output := range outputs {
+			sum += output.TxOut.Value
+		}
+		sums = append(sums, struct {
```
Comment (Beerosagos) [inline backend/coins/btc/account.go:798]: nit: I don't like it much defining the struct twice. Maybe it is better to define it once with a proper private name outside the function.

**Review by Beerosagos — APPROVED** [review]
LGTM apart for a small nit. I agree that we definitely need a proper warning banner in case of address reuse, and maybe also highlighting somehow the reused addresses (e.g. bold font, or smth else).

File: backend/coins/btc/account.go (hunk 1 above)
Comment (strmci) [reply backend/coins/btc/account.go]: I added a new type `addressOutputsSum` and used it in the function.

**Comment by strmci** [comment]
>  I agree that we definitely need a proper warning banner in case of address reuse, and maybe also highlighting somehow the reused addresses (e.g. bold font, or smth else).

Would you like to add it here, or should we do it in a separate PR?

**Comment by thisconnect** [comment]
I think separate PR makes sense or at least separate commit. For example in the case we need to undo one of the 2 changes.

**Comment by strmci** [comment]
PR to add address reuse warning: https://github.com/BitBoxSwiss/bitbox-wallet-app/pull/2724

**Pull request merged**